homeassistant/components/smarttub/config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Config flow to configure the SmartTub integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, CONF_SCAN_INTERVAL
+from homeassistant.core import callback
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN  # pylint: disable=unused-import
+from .controller import SmartTubController
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_EMAIL): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SmartTubConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """SmartTub configuration flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            controller = SmartTubController(self.hass)
+            account_id = await controller.get_account_id(
+                user_input[CONF_EMAIL], user_input[CONF_PASSWORD]
+            )
+            if account_id is not None:
+                await self.async_set_unique_id(account_id)
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(
+                    title=user_input[CONF_EMAIL], data=user_input
+                )
+
+            errors["base"] = "invalid_auth"
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_reauth(self, user_input=None):
+        """Handle a flow initiated by failed auth."""
+        return await self.async_step_user(user_input)
+
```

Comment:
This isn't actually implemented so it should be removed

Reply:
Thanks for the review. I'll remove it, but I would prefer to implement it properly if you could point me to a good example. It's not entirely clear to me how it's supposed to work.

Reply:
OK, removing it for now. If you could point me to a good example, I'll work on implementing it properly.

Reply:
Can you clarify what's not implemented? I I remove this, the test fails with:

```
ERROR    homeassistant:runner.py:93 Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/Users/zero/src/mine/home-assistant-core/homeassistant/data_entry_flow.py", line 130, in async_init
    result = await self._async_handle_step(
  File "/Users/zero/src/mine/home-assistant-core/homeassistant/data_entry_flow.py", line 208, in _async_handle_step
    raise UnknownStep(
homeassistant.data_entry_flow.UnknownStep: Handler SmartTubConfigFlow doesn't support step reauth
```

Reply:
Reauth flows need to update existing config entry data and then abort instead of creating a new entry.

There is an example in `unifi` or a simpler one in `august`.  This PR is already large so it would be better to do it in a followup.